models/DataObject/Service.php

```diff
@@ -432,7 +432,11 @@ public static function gridObjectData(AbstractObject $object, array $fields = nu
                                     }
                                 }
                             } else {
-                                $data[$dataKey] = $valueObject->value;
+                                AbstractObject::setGetInheritedValues(true);
+
+                                $data[$dataKey] = $object->get($key, $user->getLanguage());
+
+                                AbstractObject::setGetInheritedValues(false);
```

Comment:
```suggestion
                                $inheritanceBackup = AbstractObject::getGetInheritedValues();
                                AbstractObject::setGetInheritedValues(true);

                                $data[$dataKey] = $object->get($key, $user->getLanguage());

                                AbstractObject::setGetInheritedValues($inheritanceBackup);
```

I think we should ensure that getInheritedValues are set back to the original state